pkg/network/network.go

```diff
@@ -50,6 +52,23 @@ func FillDefaults(conf, previous *opv1alpha1.NetworkAddonsConfigSpec) error {
 	return nil
 }
 
+// specialCleanUp checks if there are any outdated objects and deletes them.
+// This means that the object wil be installed as new and not by the upgrade kubectl algorithm
+// The cleanUp is performed on specific obsoleted objects, so some of the cleanUp functions may be not implemented.
+func SpecialCleanUp(client k8sclient.Client) error {
```

Comment:
We are not deleting outdated objects though (in all cases at least). We use this function to overcome incompatible changes AND delete outdated stuff. e.g. In this case we perform renaming by removing and recreating.

You are right we should reflect that in the comment.

Reply:
I hate comments, they always rot.

Reply:
It is not comment of how is this function implemented but on what should go in it. It is a guidance for users on how to use this "hook point". If there will be a contribution breaking the comment/purpose, it would lead to -1.